plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -482,6 +491,14 @@ else if (precision > Decimals.MAX_PRECISION || actualPrecision <= 0) {
                         DISABLE_PUSHDOWN));
 
             case OracleTypes.TIMESTAMP:
+                // nano precision
+                if (typeHandle.getDecimalDigits().isPresent() && typeHandle.getDecimalDigits().get() == 9) {
```

Comment:
Does oracle supports only nano precision or can it be between `0` and `6`

Reply:
Oracle supports as far as I see any precision from 0 to 9.

In trino for oracle as default we have milis (3).
But we could add also micros in separate PR.

Reply:
Can we file an issue and leave TODO comment? Special handling only for nano precision is weird to me.

Reply:
Why is it not possible to do what we do in SQLServer for example where we map a `timestamp(p)` from SQL Server to matching `timestamp(p)` in Trino.

This change means Oracle connector would either use NANOS or MILLIS, no other precisions even though they seem technically possible to implement.

Reply:
will try to generify

Reply:
@ebyhr @hashhar Can you please take a look again, I refactored to any precision